### PR TITLE
refactor: #CRRE-458 removing recursive functions in orderregioncontroller

### PR DESCRIPTION
**Pull request by BastienBenard**
## Describe your changes
Removal of recursive function and limitation of use of recursive function.
Addition of model

Add dependancy:
org.reflections:reflections use for test

## Checklist tests
As admin generated a request to the bookseller. The orders that were validated were indeed sent in the cvs of the email and are no longer displayed.

## Issue ticket number and link
[CRRE-458](https://jira.support-ent.fr/browse/CRRE-458)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/fr/openent/crre/helpers/FutureHelper.java [hunk 1]
```diff
@@ -59,5 +72,62 @@ public static Handler<Either<String, JsonObject>> handlerJsonObject(Promise<Json
             }
         };
     }
+
+    public static <L, R> Handler<Either<L, R>> handlerEitherPromise(Promise<R> promise) {
+        return event -> {
+            if (event.isRight()) {
+                promise.complete(event.right().getValue());
+            } else {
+                String message = String.format("[PresencesCommon@%s::handlerEitherPromise]: %s",
```
Comment (sjollois) [inline src/main/java/fr/openent/crre/helpers/FutureHelper.java:81]: Reste de Presences

**Review by sjollois — APPROVED** [review]

File: src/main/java/fr/openent/crre/helpers/StringHelper.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        if (isNullOrEmpty(str)) {
+            return str;
+        }
+        // Empty String
+        StringBuilder result = new StringBuilder();
+
+        // Append first character(in lower case)
+        // to result string
+        char c = str.charAt(0);
+        result.append(Character.toLowerCase(c));
```
Comment (qpic-cgi) [inline src/main/java/fr/openent/crre/helpers/StringHelper.java:23]: Pour la science:
return String.format("%s%s", Character.toLowerCase(str.charAt(0)), str.substring(1).replaceAll("[A-Z ]+", String.format("_%s", Character.toLowerCase("$"))));

**Review by qpic-cgi — CHANGES_REQUESTED** [review]

File: src/main/java/fr/openent/crre/helpers/FutureHelper.java (hunk 1 above)
Comment (BastienBenard) [reply src/main/java/fr/openent/crre/helpers/FutureHelper.java:81]: Tout les Object Java possède une méthode toString. Les seuls éléments qui ne possède pas toString sont les primitives (dans notre cas ce ne peut pas être une primitive) et l'élément null. Je vais rajouter une protection contre l'élément null

File: src/main/java/fr/openent/crre/helpers/StringHelper.java (hunk 2 above)
Comment (BastienBenard) [reply src/main/java/fr/openent/crre/helpers/StringHelper.java:23]: Si tu trouve le code exacte a mettre je suis preneur

**Review by qpic-cgi — APPROVED** [review]

**Pull request merged**